### ISSUE TITLE
Doesn't work outside github pages

### ISSUE THREAD

**Issue by awoimbee** [goerz/docs_versions_menu#25]
### Description
This project doesn't work on anything that is not github pages. Even trying it out with `python -m http.server` doesn't work.

### What I Did
I forked the project because it's perfect for my use case, other than the GH pages thing.
Here is the commit that makes it work anywhere: https://github.com/awoimbee/docs_versions_menu/commit/b9f221d9ab10367fb5a6b8964ddf8a3650afe060

Are you interested in bringing some of this into this repo ?
My modifications aren't clean at all (not meant for a PR) (Also I don't like jquery nor `var` 😄 ) but I can clean them up for you if you want.

**Comment by goerz** [comment]
You are correct that out of the box, the plugin only works on Github pages. Anything else requires modifying the JS-template at the moment.

I absolutely have plans to rewrite the template so that it works under a much wider range of hosts, in a future version. This includes both the situation where the project isn't hosted on Github and where the documentation isn't hosted on Github pages (although that will remain the *primary* use case). I'll probably want to move more information into the `docs_versions_menu_conf` settings in `conf.py` so that there will be little if any need for people to touch the JS-template

How *exactly* the template can be made more robust will require some planning on my end. I might accept pull requests to improve the template in this respect, although I can't make any promises I'll have time to review or merge it in the next couple of weeks. I'll want to think a little about where I want the design of the template to go.

If you do make a pull request, please keep it minimal: don't mix in unrelated changes such as getting rid of jquery/`var`, renaming variables, or reformatting code, in the same PR. That being said, I would definitely be open to merge in PR's for such changes separately. I barely know Javascript, so I'd welcome improvements, provided they're reasonably atomic and come with a little bit of explanation: why is `let` better than `var`? Are there expected compatibility/performance improvements with getting rid of jQuery?

**Comment by awoimbee** [comment]
Ok, thanks for the quick response. I'll clean up and use my fork while you figure out where you want this project to go. I think we don't have the same use-case and I want/need fast updates.
Don't worry I will never open a PR where I half-assed re-wrote most of a file like in the cited commit 😄.

Regarding the explanations:
- `let` is better than `var` because it's _much_ stricter so less bug inducing and acts more like in other languages. Check out https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var
- jquery is bad in this day and age because it's a fat useless lib, native ES6 has _way_ better APIs. Also, `async` instead of callback hell
